presto-accumulo/src/main/java/com/facebook/presto/accumulo/metadata/AccumuloMetadataManager.java

```diff
@@ -0,0 +1,213 @@
+
+    /**
+     * Gets all view names that have the given schema. The returned table names should not contain
+     * the schema.
+     *
+     * @param schema Schema name
+     * @return Set of all view names with the given schema
+     */
+    public abstract Set<String> getViewNames(String schema);
+
+    /**
+     * Gets the {@link AccumuloView} object for the given schema and view
+     *
+     * @param table Schema and view name
+     * @return The AccumuloTable object, or null if does not exist.
+     */
+    public abstract AccumuloView getView(SchemaTableName table);
+
+    /**
+     * Creates and store table metadata for the given table
+     *
+     * @param table Table to create the metadata for
+     */
+    public abstract void createTableMetadata(AccumuloTable table);
+
+    /**
+     * Destroy the metadata for the given table
+     *
+     * @param stName Schema and table name
+     */
+    public abstract void deleteTableMetadata(SchemaTableName stName);
+
+    /**
+     * Creates and stores view metadata
+     *
+     * @param view View to create metadata for
+     */
+    public abstract void createViewMetadata(AccumuloView view);
+
+    /**
+     * Destroy the metadata for the given view
+     *
+     * @param stName Schema and view name
+     */
+    public abstract void deleteViewMetadata(SchemaTableName stName);
+
+    /**
+     * Gets a Boolean value indicating whether or not the given byte array can be deserialized to an {@link AccumuloTable} object
+     *
+     * @param data Data to check
+     * @return True if the bytes are an AccumuloTable, false otherwise
+     * @throws IOException If an IOException occurs
+     */
+    protected boolean isAccumuloTable(byte[] data)
+            throws IOException
+    {
+        // AccumuloTable does not contain a 'data' node
+        return !mapper.reader().readTree(new String(data)).has("data");
+    }
+
+    /**
+     * Gets a Boolean value indicating whether or not the given byte array can be deserialized to an {@link AccumuloView} object
+     *
+     * @param data Data to check
+     * @return True if the bytes are an AccumuloView, false otherwise
+     * @throws IOException If an IOException occurs
+     */
+    protected boolean isAccumuloView(byte[] data)
+            throws IOException
+    {
+        // AccumuloView contains a 'data' node
+        return mapper.reader().readTree(new String(data)).has("data");
+    }
+
+    /**
+     * Converts the given byte array to an {@link AccumuloTable}.
+     *
+     * @param data byte array of a serialized AccumuloTable
+     * @return AccumuloTable
+     * @throws JsonParseException If a parsing exception occurs
+     * @throws JsonMappingException If a mapping exception occurs
+     * @throws IOException If an IOException occurs
+     */
+    protected AccumuloTable toAccumuloTable(byte[] data)
+            throws JsonParseException, JsonMappingException, IOException
```

Comment:
`JsonParseException` and `JsonMappingException` both extend `IOException`, why all declared?